uint16[] quadrant_1
uint16[] quadrant_2
uint16[] quadrant_3
uint16[] quadrant_4
